android/uhabits-android/src/main/java/org/isoron/uhabits/receivers/WidgetReceiver.java

```diff
@@ -105,6 +104,9 @@ public void onReceive(final Context context, Intent intent)
                             data.getTimestamp());
                     break;
                 case ACTION_SET_NUMERICAL_VALUE:
+                    // close Notification drawer
+                    context.sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
```

Comment:
I got to the same solution when trying to account for this problem in a different PR. One thing to note is that entering multiple habits with every one closing the notification drawer becomes a bit annoying. I don't have a better solution to this than somehow inlining the answer instead of showing the dialog. I know that some chat clients open a text view inside the notification when you click "Reply", so maybe we can try a similar approach?

Reply:
The Android documentation refers to this as [direct replies](https://developer.android.com/training/notify-user/build-notification#reply-action). It may actually work well. This can be a separate PR.